File: docs/guides/swift/Sources/savePopularRecords.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

struct Record: Codable {
    let twitterHandle: String
    let nbFollowers: Int
    var isPopular = false
}

func savePopularRecords() async throws {
    do {
        let client = try SearchClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY")

        var records: [Record] = []

        try await client.browseObjects(
            indexName: "YOUR_INDEX_NAME",
            browseParams: BrowseParamsObject(),
            aggregator: { (response: BrowseResponse<Record>) in
                records.append(contentsOf: response.hits.map {
                    $0.nbFollowers < 1_000_000 ? $0 : Record(
                        twitterHandle: $0.twitterHandle,
                        nbFollowers: $0.nbFollowers,
                        isPopular: true
                    )
                })
            }
        )

        try await client.saveObjects(indexName: "<YOUR_INDEX_NAME>", objects: records)
    } catch {
        print(error)
    }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/savePopularRecords.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.JsonPrimitive
import kotlinx.serialization.json.buildJsonObject

suspend fun savePopularRecords() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  var records: List<JsonObject> = listOf()

  client.browseObjects(
    "YOUR_INDEX_NAME",
    BrowseParamsObject(),
    aggregator = { response ->
      records =
        records +
          response.hits.map { hit ->
            val props = hit.additionalProperties ?: mapOf()

            val nbFollowers = props["nbFollowers"].toString().toInt()

            buildJsonObject {
              put("twitterHandle", JsonPrimitive(props["twitterHandle"].toString()))
              put("nbFollowers", JsonPrimitive(nbFollowers))
              put("isPopular", JsonPrimitive(nbFollowers > 1000000))
            }
          }
    },
  )
  client.saveObjects(indexName = "<YOUR_INDEX_NAME>", objects = records)
}
